keps/1282-pods-ready-requeue-strategy/README.md

```diff
@@ -208,6 +208,22 @@ type RequeueState struct {
 }
 ```
 
+Add a new field `waitForPodsReadyStats` to the WorkloadStatus struct to track the number of evictions by reason:
+
+```go
+type WorkloadStatus struct {
+    // waitForPodsReadyStats holds statistics for WaitForPodsReady requeues.
+    // +optional
+    WaitForPodsReadyStats *WaitForPodsReadyStats `json:"waitForPodsReadyStats,omitempty"`
+}
+
+type WaitForPodsReadyStats struct {
+    // evictionByReasonCount maps eviction reasons to their counts.
+    // +optional
+    EvictionByReasonCount map[string]int32 `json:"evictionByReasonCount,omitempty"`
```

Comment:
I still do not understand what is relationship between existing requeueState.

Howeve, at least, I can say this should not be introduced since map will lose any kind of CRD based possibilities.
If we want to introduce such APIs, we should use a dedicated struct having reason and count.

Reply:
I'm find with count and reason. Maybe more generic

evictionStats:
   waitForPodsReady:
    - reason: StartupTimeout 
      count: 5
    - reason: RecoveryTimeout
      count: 2

Reply:
regarding the requeueState, ok n the current form it does not carry the required information currently. It is just a counter without reason, counting not only waitforpodsready

Reply:
What do you think about this?

```
evictionStats:
    - type: PodsReadyTimeout 
      reason: StartupTimeout 
      count: 5
    - type: PodsReadyTimeout 
      reason: RecoveryTimeout
      count: 2
```

Reply:
sgtn

Reply:
And to be fair I like this proposal the most so far

Reply:
Updated